src/Management/src/Endpoint/Actuators/Health/HealthEndpointHandler.cs

```diff
@@ -96,56 +77,76 @@ public async Task<HealthEndpointResponse> InvokeAsync(HealthEndpointRequest heal
         return response;
     }
 
-    private ICollection<HealthCheckRegistration> GetFilteredHealthCheckServiceOptions(string requestedGroup)
+    private async Task<HealthEndpointResponse> GetResponseAsync(HealthGroupOptions? groupOptions, CancellationToken cancellationToken)
     {
-        if (!string.IsNullOrEmpty(requestedGroup))
-        {
-            if (_endpointOptionsMonitor.CurrentValue.Groups.TryGetValue(requestedGroup, out HealthGroupOptions? groupOptions) && groupOptions.Include != null)
-            {
-                string[] includedContributors = groupOptions.Include.Split(',');
+        IHealthContributor[] filteredContributors = GetFilteredContributors(groupOptions);
+        ICollection<HealthCheckRegistration> healthCheckRegistrations = GetFilteredRegistrations(groupOptions);
 
-                return _healthOptionsMonitor.CurrentValue.Registrations
-                    .Where(registration => includedContributors.Contains(registration.Name, StringComparer.OrdinalIgnoreCase)).ToArray();
-            }
+        HealthCheckResult result = await _healthAggregator.AggregateAsync(filteredContributors, healthCheckRegistrations, _serviceProvider, cancellationToken);
 
-            _logger.LogInformation("Health check requested for a group that is not configured");
-        }
-        else
+        return new HealthEndpointResponse(result);
+    }
+
+    private ICollection<HealthCheckRegistration> GetFilteredRegistrations(HealthGroupOptions? groupOptions)
```

Comment:
Variables in this method are named around "contributors", but should be "registrations" instead.